.github/workflows/e2e-android-detox.yml

```diff
@@ -0,0 +1,80 @@
+name: Detox E2E Android Tests PR
+
+on:
+    pull_request:
+      branches:
+        - main
+      types:
+        - labeled
+
+concurrency:
+    group: "${{ github.workflow }}-${{ github.event.pull_request.number }}-${{ github.event.label.name }}"
+    cancel-in-progress: true
+
+jobs:
+    build-android-apk:
+        if: github.event.label.name == 'E2E Android tests for PR'
+        runs-on: ubuntu-latest-8-cores
```

Comment:
Looking at the existing runners, we're not using large but standard size only. Could you check if `ubuntu-22.04` would be enough?